### DISCUSSION TITLE
How can I extract node text from the ydoc tiptap update server-side?

### DISCUSSION THREAD

**Discussion by sradu** [ueberdosis/hocuspocus#576] [Q&A]
I'm using the hocuspocus to save a tiptap document via the standard extension-database (which is awesome!).

```
store: async ({ documentName, state }) => { 
   client.query('UPDATE "pages" SET content = $2 WHERE id = $1', [documentName, state], (error, res) => {
});
```

What I'd like to do is extract from the document state the first node and save it in the db separately as a title. So my tiptap content might be something like `<><heading>Header</heading>....</>`, I want to save `Header` as a separate column.

How can I parse `state` server-side/node?

**Comment by yan-yanishevsky** [comment]
https://github.com/yjs/y-prosemirror/ library has `yDocToProsemirrorJSON` utility function that might help

```js
import { yDocToProsemirrorJSON } from 'y-prosemirror'
// ...

store: async ({ document, documentName, state }) => { 
  const json = yDocToProsemirrorJSON(document, 'default')
  // ...
});
```
